datafusion/core/src/execution/memory_manager.rs

```diff
@@ -342,8 +349,8 @@ impl MemoryManager {
                 // if we cannot acquire at lease 1/2n memory, just wait for others
                 // to spill instead spill self frequently with limited total mem
                 debug!(
-                    "Cannot acquire minimum amount of memory {} on memory manager {}, waiting for others to spill ...",
-                    human_readable_size(min_per_rqt), self);
+                    "Cannot acquire a minimum amount of {} memory from the manager of total {}, waiting for others to spill ...",
```

Comment:
The second fix: stop calling std::fmt::Display on memory manager since it's acquiring a lock we are already holding.